ynr/apps/candidates/models/versions.py

```diff
@@ -160,19 +160,8 @@ def revert_person_from_version_data(person, version_data, part_of_merge=False):
         )
 
     # Remove all candidacies, and recreate:
-    for membership in Membership.objects.filter(
-        person=person,
-        role=F("post_election__election__candidate_membership_role"),
-    ):
-        # At the moment the merge code has its own way of preserving
-        # the uk_results CandidateResult data (see
-        # additional_merge_actions), so they will be
-        # recreated. (FIXME: omitting this check when merging does
-        # mean that we're not checking for other models in the future
-        # that may have a foreign key to Membership when doing
-        # merges.)
-        if not part_of_merge:
-            raise_if_unsafe_to_delete(membership)
+    for membership in Membership.objects.filter(person=person):
+        raise_if_unsafe_to_delete(membership)
```

Comment:
This is a bit of a tangent, but while I notice it..

`raise_if_unsafe_to_delete()` relies on `paired_object_safe_to_delete()`:

https://github.com/DemocracyClub/yournextrepresentative/blob/c700992cc98eae37609457de6c4b3598c199dd88/ynr/apps/candidates/models/popolo_extra.py#L36-L60

Now that we've removed all (I _think_ ) of the paired models, do we need to review this?

Reply:
In e17651699ec53007e50acc99ad601e8255d3af3e I've renamed `paired_object_safe_to_delete` to `model_has_related_objects`, but also made #782 to deal with this properly.

Reply:
OK. Renaming it is fine, but isn't the actual logic of that function allowing us to delete more than one model object ( `if len(collected[1]) == 1` ) under the assumption its ok to delete a model and its corresponding `.extra` ?
I don't think that's the right check any more, is it, or have I misunderstood?

Reply:
Right, no you're right. I've changed the calls and written a couple of simple tests for it now.